### PR TITLE
add our own head() method so relative URLs work for HEAD requests

### PR DESCRIPTION
**Pull request by simbabque**
Also fixes some POD formatting in the vicinity.

This came out of https://stackoverflow.com/a/68635803/1331451.

### CODE REVIEW

**Comment by simbabque** [comment]
I've raised #322 because of this test failure. Tried to restart, that didn't work, then ran it locally.

File: lib/WWW/Mechanize.pm [hunk 1]
```diff
@@ -510,6 +513,29 @@ sub _SUPER_put {
...
+=head2 $mech->head ($uri )
+
+Performs a HEAD request to I<$uri>. Returns an L<HTTP::Response> object.
+I<$uri> can be a well-formed URI string, a L<URI> object, or a
+L<WWW::Mechanize::Link> object.
+
+=cut
+
+sub head {
+   my $self = shift;
```
Comment (oalders) [inline lib/WWW/Mechanize.pm:525]: Looks like the spacing is off by one here? Maybe we could just run the new code through `perltidy`?

File: t/local/head.t [hunk 2]
```diff
@@ -0,0 +1,31 @@
+use warnings;
+use strict;
+use Test::More tests => 8;
```
Comment (oalders) [inline t/local/head.t:3]: I think you're following the pattern of other tests, but I don't think we need to do that. Using `done_testing()` should be good enough.

File: t/local/head.t [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+
+BEGIN {
+    use Tools;
+}
+
+BEGIN {
+    delete @ENV{ qw( IFS CDPATH ENV BASH_ENV ) };
+    use_ok( 'WWW::Mechanize' );
+}
+
```
Comment (oalders) [inline t/local/head.t:16]: I don't think `use Tools` is doing much here. It looks like it does ` delete @ENV{ qw( http_proxy HTTP_PROXY PATH IFS CDPATH ENV BASH_ENV) };` and then exports `$canTMC memory_cycle_ok`, which aren't being used in this test.

**Review by oalders — CHANGES_REQUESTED** [review]
Thanks @simbabque! This is a nice improvement. Just had a few comments.

**Comment by oalders** [comment]
If you rebase, those test failures should be resolved.

File: t/local/head.t (hunk 3 above)
Comment (simbabque) [reply t/local/head.t:16]: I've had a look at other tests, and there's [a comment in add_header.t for the `delete` line](https://github.com/libwww-perl/WWW-Mechanize/blob/900faaf6e64c6ab60cc4c6188c070b8f29b0776f/t/add_header.t#L9) that says _Placates taint-unsafe Cwd.pm in 5.6.1_. Actually it seems like nearly all the tests that actually run `get()` delete these environment variables. 

I agree we don't need `use Tools`.

File: t/local/head.t (hunk 2 above)
Comment (simbabque) [reply t/local/head.t]: I had that first, then undid it because of the old tests. Probably about time we clean those up.

**Comment by simbabque** [comment]
All cleaned up. Would you mind pushing a release when this has landed please?

**Review by oalders — APPROVED** [review]
Thanks, @simbabque!

**Comment by oalders** [comment]
2.04 has now been released.

**Comment by simbabque** [comment]
Thank you for the speedy release! :up:

**Pull request merged**